Dockerfile-Client

```diff
@@ -10,11 +10,21 @@ RUN rm -rf /code/dist \
     && python setup.py sdist \
     && mv /code/dist/$(ls /code/dist | head -1) /code/dist/gordo-components-packed.tar.gz
 
+# Extract a few big dependencies which docker will cache even when other dependencies change
+RUN cat /code/requirements.txt | grep tensorflow== > /code/prereq.txt \
+    && cat /code/requirements.txt | grep pyarrow== >> /code/prereq.txt \
+    && cat /code/requirements.txt | grep scipy== >> /code/prereq.txt \
+    && cat /code/requirements.txt | grep catboost== >> /code/prereq.txt
+
 FROM python:3.6.9-slim-stretch
 
 # Install requirements separately for improved docker caching
+COPY --from=builder /code/prereq.txt .
+RUN pip install --no-deps -r prereq.txt --no-cache-dir
```

Comment:
Why `--no-deps` ?

Reply:
Becasue we dont want this step to pull inn (possibly unpinned) dependencies for these things. With this way here we are downloading and installing only these packages, and then installing the rest afterwards. 

We *could* drop --no-deps, but then things would get interesting in `pip install -r /code/requirements.txt`, since it would then potentially have to uninstal the installed-but-wrong version of something.

Reply:
So flink, you've thought of everything. :+1: